libs/@blockprotocol/graph/src/stdlib/subgraph/edge/link-entity.ts

```diff
@@ -62,46 +73,74 @@ export const getOutgoingLinksForEntity = (
 
   return (
     Object.entries(entityEdges)
-      // Only look at outgoing edges that were created before or at the timestamp
+      // Only look at outgoing edges that were created before or within the search interval
       .filter(
-        ([edgeTimestamp, _outwardEdges]) => edgeTimestamp <= timestampString,
+        ([edgeTimestamp, _outwardEdges]) =>
+          compareBounds(
+            { kind: "inclusive", limit: edgeTimestamp },
+            searchInterval.end,
+            /** @todo - should this be end */
+            "start",
```

Comment:
Yes, I think so. As `searchInterval.end` could be `"exclusive"`, this would return `true` if `searchInterval.limit === edgeTimestamp` if I got it correctly.

Reply:
Looking at the implementation of `compareBounds` I'm not convinced it actually changes things. 

But considering this is basically half of an `intervalOverlapsInterval` operation, I now think maybe it _should_ be `end`? It's effectively this portion (I think)

<img width="802" alt="image" src="https://user-images.githubusercontent.com/25749103/216310305-a6ac44d9-a661-42d9-8480-2a792eaa250c.png">

Reply:
(From your [other comment](https://github.com/blockprotocol/blockprotocol/pull/928#discussion_r1093565147))

> Maybe we should add a function `boundIsStrictlyRightOfInterval` and use that instead.

This sounds sensible. Do we want `strictlyRightOf` here? I'd assume we're also happy with the bounds being the same (e.g. inclusive at timestamp 3, and inclusive at timestamp 3)

Reply:
For an "exclusive" this will change from "is smaller" to "is equal" but as we compare with `<=` both should be fine, yes. 

Yes, let's add a `strictlyRightOf`-like function, it makes reading it easier.

Reply:
Sorry I wasn't clear, what I meant by

> Do we want `strictlyRightOf` here?

Is that the operation we're doing doesn't _seem_ to be **strictly right of**.

Reply:
No, we would check, if the interval is `not strictlyRightOf` I think 🙂

Reply:
And actually we're not checking if the bound is to the right of the interval. We're checking if the bound is to the left of or contained in the interval

Reply:
Yes, this is the same as if the bound is not strictly right of the interval

Reply:
(We talked about this offline and landed on this) https://github.com/blockprotocol/blockprotocol/pull/928/commits/3a11838b5d74494484778721270590f661a70c30